src/metrics.ts

```diff
@@ -0,0 +1,64 @@
+import { collectDefaultMetrics, Counter, Gauge, register } from 'prom-client';
+
+if (process.env.NODE_ENV == 'production') {
+  register.clear();
+  collectDefaultMetrics({ prefix: 'peribolos_' });
+}
+
+export const totalNumOfReposServing =
+  (register.getSingleMetric(
+    'peribolos_total_num_of_repos_serving'
+  ) as Gauge<string>) ||
+  new Gauge({
+    name: 'peribolos_total_num_of_repos_serving',
+    help: 'Total numbers of repositories serving.',
+    labelNames: [],
+  });
+
+export const numberOfInstallTotal =
+  (register.getSingleMetric(
+    'peribolos_num_of_install_total'
+  ) as Counter<string>) ||
+  new Counter({
+    name: 'peribolos_num_of_install_total',
+    help: 'Total number of installs received',
+    labelNames: [],
+  });
+
+export const numberOfUninstallTotal =
+  (register.getSingleMetric(
+    'peribolos_num_of_uninstall_total'
+  ) as Counter<string>) ||
+  new Counter({
+    name: 'peribolos_num_of_uninstall_total',
+    help: 'Total number of uninstalls received',
+    labelNames: [],
+  });
+
+export const numberOfActionsTotal =
+  (register.getSingleMetric(
+    'peribolos_num_of_actions_total'
+  ) as Counter<string>) ||
+  new Counter({
+    name: 'peribolos_num_of_actions_total',
+    help: 'Total number of actions received',
+    labelNames: ['install', 'action'],
+  });
+
+export const operationResults =
+  (register.getSingleMetric(
+    'peribolos_operations_results'
+  ) as Counter<string>) ||
+  new Counter({
+    name: 'peribolos_operations_results',
+    help: 'Peribolos operations results',
```

Comment:
We should note that this doesn't ensure the operation was success. We don't track if Tekton succeeded in performing the task. We can only track if we instantiated the taskrun successfully. The rest needs to be derived from tekton metrics. Can you maybe adjust the description to match the data better, please?